app/data_client.go

```diff
@@ -320,22 +346,23 @@ func BsonToGo(rawData [][]byte) ([]map[string]interface{}, error) {
 }
 
 // TabularDataByFilter queries tabular data and metadata based on given filters.
-func (d *DataClient) TabularDataByFilter(
-	ctx context.Context,
-	filter Filter,
-	limit int,
-	last string,
-	sortOrder Order,
-	countOnly bool,
-	includeInternalData bool,
-) (TabularDataReturn, error) {
+func (d *DataClient) TabularDataByFilter(ctx context.Context, opts *DataByFilterOptions) (TabularDataReturn, error) {
+	dataReq := pb.DataRequest{}
+	var countOnly, includeInternalData bool
+	if opts != nil {
+		dataReq.Filter = filterToProto(opts.Filter)
+		if opts.Limit != 0 {
```

Comment:
If `opts.Limit` is not specified, should we default it to 50 somewhere in this function? also, how do we differentiate between a limit explicitly set to 0 and one that is not set (given that the default value for integers in Go is 0, iirc)?

Reply:
This is something I know from my work in the Python SDK. The default to 50 happens server-side